### PR TITLE
test: refactor random tests to use validator.js methods

### PR DESCRIPTION
**Pull request by import-brain**

### CODE REVIEW

File: test/random.spec.ts [hunk 1]
```diff
@@ -257,9 +257,7 @@ describe('random', () => {
       });
 
       expect(alphaText).toHaveLength(5);
-      for (const bannedChar of bannedChars) {
-        expect(alphaText).not.includes(bannedChar);
-      }
+      expect(alphaText).toMatch(/^[0-9]*$/);
```
Comment (import-brain) [inline test/random.spec.ts:260]: This matches numbers only.

File: test/random.spec.ts [hunk 2]
```diff
@@ -269,9 +267,7 @@ describe('random', () => {
       });
 
       expect(alphaText).toHaveLength(5);
-      for (const bannedChar of bannedChars) {
-        expect(alphaText).not.includes(bannedChar);
-      }
+      expect(alphaText).toMatch(/^([^0-9]*)$/);
```
Comment (import-brain) [inline test/random.spec.ts:270]: This matches everything except numbers.

File: test/random.spec.ts (hunk 1 above)
Comment (Shinigami92) [inline test/random.spec.ts:260]: While reading this test, I find it confusing that the variable is called `alphaText`. I assume this is a copy-paste issue from history. Should we name it maybe `actual` or `alphaNumericText`?
I would prefer `actual` for simplicity.

Applies for the part below as well.

File: test/random.spec.ts (hunk 2 above)
Comment (Shinigami92) [reply test/random.spec.ts:270]: Should we use `\d` for this?

File: test/random.spec.ts (hunk 1 above)
Comment (Shinigami92) [reply test/random.spec.ts:260]: Should we use `\d` for this?

**Comment by xDivisionByZerox** [comment]
I don't know about this PR. If I remember correctly we have an unwritten rule to use as less RegExp as possible.
It can get really fast out of hand and is for most people not that readable. Furthermore (in this specific case) using the provided matcher (`not.includes`) you receive a much nicer error message to debug.

**Comment by Shinigami92** [comment]
> I don't know about this PR. If I remember correctly we have an unwritten rule to use as less RegExp as possible. It can get really fast out of hand and is for most people not that readable. Furthermore (in this specific case) using the provided matcher (`not.includes`) you receive a much nicer error message to debug.

Would you like to add this statement to our docs? And yes I also se it like you and I already thought about to suggest using `toSatisfy` with validators. But if you even say that the output message is better readable, then we may even not do anything here.

File: test/random.spec.ts (hunk 2 above)
Comment (ST-DDT) [reply test/random.spec.ts:270]:
```suggestion
      expect(alphaText).toMatch(/^[a-z]*$/);
```

File: test/random.spec.ts (hunk 1 above)
Comment (ST-DDT) [reply test/random.spec.ts:260]: I think this is good enough.
If we rename the variables, then we should do so consistently in another PR.

**Comment by ST-DDT** [comment]
Maybe instead of regex use the related validator.js methods in compination with satisfy?

File: test/random.spec.ts (hunk 2 above)
Comment (Shinigami92) [reply test/random.spec.ts:270]: Are upper case letters also allowed? Or is that another case not covered in this test-path?

**Comment by import-brain** [comment]
> Maybe instead of regex use the related validator.js methods in compination with satisfy?

I didn't use satisfy directly, but I think my method also works. There might be a better way to write this, though.

**Review by ST-DDT — APPROVED** [review]

**Review by ST-DDT — CHANGES_REQUESTED** [review]
Oh no. Now the failure message no longer contains the bad value.

**Review by Shinigami92 — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Pull request closed without merging**